pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+		// update the quotaNode's quotaInfo
+		updateQuotaInfo.UpdateQuotaInfoFromRemote(quotaInfo)
+	}
+
+	// build tree according to the parGroupName
+	for _, topoNode := range allQuotaTopoNodes {
+		if topoNode.groupName == extension.RootQuotaName {
+			continue
+		}
+		parQuotaTopoNode := allQuotaTopoNodes[topoNode.quotaInfo.ParentName]
+		topoNode.parQuotaTopoNode = parQuotaTopoNode
+		parQuotaTopoNode.AddChildGroupQuotaInfo(topoNode)
+	}
+
+	return allQuotaTopoNodes
+}
+
+// ResetAllGroupQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) resetAllGroupQuotaNoLock(newAllQuotaTopoNodes map[string]*QuotaTopoNode) {
+	newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap, childRequestMap, childUsedMap := make(quotaResMapType), make(quotaResMapType),
+		make(quotaResMapType), make(quotaResMapType), make(quotaResMapType)
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		if quotaName == extension.RootQuotaName {
+			continue
+		}
+		topoNode.quotaInfo.lock.Lock()
+		newMaxQuotaMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Max.DeepCopy()
+		newMinQuotaMap[quotaName] = topoNode.quotaInfo.CalculateInfo.OriginalMin.DeepCopy()
+		newSharedWeightMap[quotaName] = topoNode.quotaInfo.CalculateInfo.SharedWeight.DeepCopy()
+
+		topoNode.quotaInfo.setMaxQuotaNoLock(v1.ResourceList{})
+		topoNode.quotaInfo.setOriginalMinQuotaNoLock(v1.ResourceList{})
+		topoNode.quotaInfo.setSharedWeightNoLock(v1.ResourceList{})
+
+		if !topoNode.quotaInfo.IsParent {
+			childRequestMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Request.DeepCopy()
+			childUsedMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Used.DeepCopy()
+		}
+
+		topoNode.quotaInfo.CalculateInfo.Request = v1.ResourceList{}
+		topoNode.quotaInfo.CalculateInfo.Used = v1.ResourceList{}
+		topoNode.quotaInfo.RuntimeVersion = 0
+		topoNode.quotaInfo.CalculateInfo.Runtime = v1.ResourceList{}
+		topoNode.quotaInfo.lock.Unlock()
+	}
+
+	// store Information of SystemQuotaGroup and DefaultQuotaGroup
+	defaultInfo := gqm.quotaInfoMap[extension.DefaultQuotaName].DeepCopy()
+	systemInfo := gqm.quotaInfoMap[extension.SystemQuotaName].DeepCopy()
+
+	// clear old structure
+	gqm.quotaInfoMap = make(map[string]*QuotaInfo)
+	gqm.quotaTreeWrapperMap = make(map[string]*QuotaTreeWrapper)
+	gqm.quotaTopoNode = make(map[string]*QuotaTopoNode)
+
+	// reset structure
+	gqm.quotaInfoMap[extension.DefaultQuotaName] = defaultInfo
+	gqm.quotaInfoMap[extension.SystemQuotaName] = systemInfo
+	gqm.quotaTreeWrapperMap[extension.RootQuotaName] = NewQuotaTreeWrapper(extension.RootQuotaName)
+	gqm.quotaTreeWrapperMap[extension.RootQuotaName].SetClusterTotalResource(gqm.totalResourceIndeed)
+	rootNode := newAllQuotaTopoNodes[extension.RootQuotaName]
+	gqm.resetAllGroupQuotaRecursiveNoLock(rootNode, newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap)
+
+	// subGroup's topo relation may change; refresh the request/used from bottom to up
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
```

Comment:
actually newAllQuotaTopoNodes is just gqm.quotaTopoNode itself.... no need to clear gqm.quotaTopoNode